crates/wasmparser/src/validator/operators.rs

```diff
@@ -1416,12 +1416,46 @@ where
         }
     }
 
+    fn cont_type_at(&self, at: u32) -> Result<&ContType> {
+        let sub_ty = self.sub_type_at(at)?;
+        if let CompositeInnerType::Cont(cont_ty) = &sub_ty.composite_type.inner {
+            if self.inner.shared && !sub_ty.composite_type.shared {
+                bail!(
+                    self.offset,
+                    "shared continuations cannot access unshared continuations",
+                );
+            }
+            Ok(cont_ty)
+        } else {
+            bail!(self.offset, "non-continuation type {at}",)
+        }
+    }
+
+    fn func_type_of_cont_type(&self, cont_ty: &ContType) -> Result<&'resources FuncType> {
+        let func_id = cont_ty.0.as_core_type_id().expect("valid core type id");
+        Ok(self.resources.sub_type_at_id(func_id).unwrap_func())
+    }
```

Comment:
This looks like it should be possible to remove the `Result` return value and just return `&'resources FuncType` here